src/PopoverMenu.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { findDOMNode } from 'react-dom';
+import cx from 'classnames';
+
+/**
+ * @module PopoverMenu
+ */
+class PopoverMenu extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.selectedIndex = 0;
+		this.updateFocusBy = this.updateFocusBy.bind(this);
+		this.handleKeyUp = this.handleKeyUp.bind(this);
+	}
+
+	updateFocusBy(delta) {
+		const menuItems = findDOMNode(this).querySelectorAll('.popover-menu-item');
+		const targetIndex = this.selectedIndex + delta;
+
+		if (menuItems[targetIndex] == undefined) {
+			return;
+		}
+
+		this.selectedIndex = targetIndex;
+		menuItems[targetIndex].focus();
```

Comment:
I think `ref` is what I was conceptually overlooking... The above changes make sense to me.